src/rf/apps/home/forms.py

```diff
@@ -48,24 +48,18 @@ def clean(self):
         except:
             self.cleaned_data['images'] = []
 
-        try:
-            start = self.cleaned_data['capture_start']
-        except KeyError:
-            # Improperly formed date string.\
-            start = None
-            self.add_error('capture_start',
-                           'The date is not valid or is not a valid format.')
+        start = self.cleaned_data.get('capture_start', None)
+        end = self.cleaned_data.get('capture_end', None)
 
-        try:
-            end = self.cleaned_data['capture_end']
-        except KeyError:
-            # Improperly formed date strings.
-            end = None
+        if start and not end:
             self.add_error('capture_end',
-                           'The date is not valid or is not a valid format.')
+                           'The date needs to be provided.')
+        if end and not start:
+            self.add_error('capture_start',
+                           'The date needs to be provided.')
```

Comment:
This is one of those rare cases where `xor` makes sense... but I'm not sure if it's a good idea.